src/node/snapshot_serialise.h

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ds/logger.h"
+#include "ds/serialized.h"
+#include "entities.h"
+#include "kv/kv_types.h"
+#include "kv/serialised_entry_format.h"
+#include "node/nodes.h"
+#include "node/service.h"
+#include "node/tx_receipt.h"
+
+#include <nlohmann/json.hpp>
+
+namespace ccf
+{
+  struct StartupSnapshotInfo
```

Comment:
`StartupSnapshotInfo` used to be in `node_state.h` and I moved it to this new file which handles snapshot serialisation/deserialisation and receipt verification.